src/librustc/traits/select.rs

```diff
@@ -3249,7 +3247,7 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
         obligation_trait_ref: ty::PolyTraitRef<'tcx>,
         expected_trait_ref: ty::PolyTraitRef<'tcx>,
     ) -> Result<Vec<PredicateObligation<'tcx>>, SelectionError<'tcx>> {
-        let obligation_trait_ref = obligation_trait_ref.clone();
+        let obligation_trait_ref = obligation_trait_ref;
```

Comment:
Am I blind or is this assignment completely unnecessary?

Reply:
Indeed, fixed, thanks!